ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala

```diff
@@ -603,15 +663,16 @@ class TransactionService(session: LedgerSession) extends LedgerTestSuite(session
     LedgerTest("TXMultiActorChoiceOk", "Accept exercising a well-authorized multi-actor choice") {
       context =>
         for {
-          ledger <- context.participant()
-          Vector(operator, receiver, giver) <- ledger.allocateParties(3)
-          agreementFactory <- ledger.create(giver, AgreementFactory(receiver, giver))
-          agreement <- ledger.exerciseAndGetContract[Agreement](
+          Vector(alpha, beta) <- context.participants(2)
+          Vector(operator, receiver) <- alpha.allocateParties(2)
+          giver <- beta.allocateParty()
+          agreementFactory <- beta.create(giver, AgreementFactory(receiver, giver))
+          agreement <- alpha.exerciseAndGetContract[Agreement](
             receiver,
             agreementFactory.exerciseAgreementFactoryAccept)
           triProposalTemplate = TriProposal(operator, receiver, giver)
-          triProposal <- ledger.create(operator, triProposalTemplate)
-          tree <- ledger.exercise(giver, agreement.exerciseAcceptTriProposal(_, triProposal))
+          triProposal <- alpha.create(operator, triProposalTemplate)
+          tree <- beta.exercise(giver, agreement.exerciseAcceptTriProposal(_, triProposal))
```

Comment:
Since `triProposal` is created on `alpha`, there is potential for flakiness when exercising `AcceptTriProposal` on `beta` if the `triProposal` contract hasn't made it to `beta` yet. Wrap in `assertEventually`?

Reply:
Definitely, I was looking only at explicit transaction reads for source of potential flakiness but I definitely overlooked this (and others, as you pointed out).

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2900/commits/b218310180d117e637a94c4e6506785a46316153